test/app-tests/loading-test.tsx

```diff
@@ -630,9 +630,18 @@ describe("loading:", function () {
                 device_id: "DEVICE_ID",
                 access_token: "access_token",
             });
-        fireEvent.change(matrixChat.container.querySelector("#mx_LoginForm_username")!, { target: { value: "user" } });
-        fireEvent.change(matrixChat.container.querySelector("#mx_LoginForm_password")!, { target: { value: "pass" } });
-        fireEvent.click(screen.getByText("Sign in", { selector: ".mx_Login_submit" }));
+        const username = matrixChat.container.querySelector("#mx_LoginForm_username");
+        if (username) {
+            fireEvent.change(username, { target: { value: "user" } });
+        }
+        const password = matrixChat.container.querySelector("#mx_LoginForm_password");
+        if (password) {
+            fireEvent.change(password, { target: { value: "pass" } });
+        }
+        const signin = screen.getByText("Sign in", { selector: ".mx_Login_submit" });
+        if (signin) {
+            fireEvent.click(signin);
+        }
```

Comment:
If the fields aren't found, the login won't work either. So bypassing that won't really work. But it'd be good if we'd get better error messages.

`screen.getByText` by definition never returns null/undefined, if the element can't be found it just throws a descriptive and clean error to explain what went wrong.

It'd be awesome if we could use the same for the other elements, something more like

```suggestion
          const usernameField = screen.getByLabelText("Username", { selector: "#mx_LoginForm_username" });
        const passwordField = screen.getByLabelText("Password", { selector: "#mx_LoginForm_password" });
        fireEvent.change(usernameField, { target: { value: "user" } });
        fireEvent.change(passwordField, { target: { value: "pass" } });
        const loginButton = screen.getByText("Sign in", { selector: ".mx_Login_submit" });
        fireEvent.click(loginButton);
```

That said, this would still fail if for example the fields take a while to load before they show up. In that case, `await screen.findByLabelText` would be more appropriate.